### DISCUSSION TITLE
How to trigger the `go` install in the renovate container

### DISCUSSION THREAD

**Discussion by mtardy** [renovatebot/renovate#23485] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.14.0

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I use a `postUpgradeTasks` with `go mod vendor` and I did not realize that in some case, depending of the order of the treatment of the updates, `go` can be not installed in the renovate container:
```
    // Upgrade the Go module directive, see:
    // https://docs.renovatebot.com/modules/versioning/#go-modules-directive-versioning
    // depName is 'go' so it will be group with the 'Go' groupName
    {
      "matchDatasources": ["golang-version"],
      "rangeStrategy": "bump",
    },
    {
      "groupName": "Go",
      "matchDepNames": [
        "go",
        "docker.io/library/golang"
      ],
      // postUpgradeTasks is only for when the Go module directives are bumped
      "postUpgradeTasks": {
        "commands": ["go mod vendor"],
        "fileFilters": ["**/**"],
        "executionMode": "branch"
      },
    },
```

I had this weird situation when I created two PRs at the same time, one had `go` installed and another did not:
- the one that had not `go` installed https://github.com/cilium/tetragon/pull/1260
- the one that had `go` installed https://github.com/cilium/tetragon/pull/1261

Here is the config https://github.com/cilium/tetragon/blob/main/.github/renovate.json5

Is there a way to force trigger the Go install how do I have to do it in the early `postUpgradeTasks` manually?

### What you tried so far.

What should I do to force renovate to install go? Is this the only option using a `postUpdateOptions` with `gomodTidy` even if it's useless? I had this in my global config and it seemed to not work -> https://github.com/cilium/tetragon/blob/main/.github/renovate.json5#L38-L40.

### Relevant debug logs

<details><summary>Logs</summary>

Unfortunately, the run that triggered this race was not run with debug logs... But I have this run where I see when Go is installed in the container https://github.com/cilium/tetragon/actions/runs/5613202960/job/15208704603#step:8:7412.
</details>

**Comment by rarkins** [comment]
Try something like `install-tool go 1.20.0`

**Reply by mtardy** [reply, marked as answer]
Thanks a lot, it was `install-tool golang 1.20.0` if anyone looks at this after.

**Reply by mtardy** [reply]
I have been trying to auto-update that version with a regex matcher only to understand after one hour that I think the json5 comments are removed so I can't really match a regex on comments in this. Is this correct @rarkins that you can't auto-update renovates' own configuration because of comments being stripped?

**Reply by rarkins** [reply]
A regex manager won't strip anything

**Reply by mtardy** [reply]
By the way, it's easier to have something like this directly and rely on the version in the main `go.mod` for people looking for this solution: `install-tool golang $(grep -oP '^go \\K.+' go.mod)`, you can use `^install-tool golang \\$\\(grep -oP '\\^go \\\\K\\.\\+' go\\.mod\\)$` for the `allowedPostUpgradeCommands` string.
